### PR TITLE
Feature/SS-6: Provision SMS authentication

### PR DESCRIPTION
**Pull request by dubstep-warrior**
**Summary**
This pull request introduces SMS authentication to the SmashSync project. It includes an authentication provider interface and implementation using the Twilio client, an authentication use case struct providing functionalities to retrieve and authenticate OTPs, two handler endpoints for these use cases, generated Swagger API specifications, and generated mocks for unit testing.

**Motivation**
To enhance security and provide a seamless user experience, implementing SMS authentication is crucial. This ensures that users can authenticate via OTPs sent to their phone numbers. Additionally, generating JWTs upon successful authentication will help maintain secure sessions.

**Changes**
- Authentication Provider Interface & Implementation: Provides SMS authentication via the Twilio client.
- Authentication Use Case Struct: Includes functionalities to retrieve and authenticate OTPs, returning a JWT token on successful authentication.
- Authentication Handler Endpoints: Corresponding handler end points for authentication use cases.
- Swagger API Specs: Generated API documentation for Auth handlers using Swagger.
- Generated Mocks: Mocks for EchoContext, Logger, SMSProvider, and UserRepository to facilitate isolated unit testing.

### CODE REVIEW

File: smashsync-go/cmd/rest/main.go [hunk 1]
```diff
@@ -1,15 +1,108 @@
...
+	// Externals
+	db           *mongoQuerier.MongoQuerier
+	twilioClient *twiliogo.RestClient
+
+	// Usecases
+	authUseCase *userUC.AuthUseCase
+	userUseCase *userMongo.MongoDBUserRepository
+)
+
+func init() {
```
Comment (YeongZhenHong) [inline smashsync-go/cmd/rest/main.go:32]: Missing comments

File: smashsync-go/cmd/rest/main.go [hunk 2]
```diff
@@ -1,15 +1,108 @@
...
+
+	// User use case
+	userRepo := userMongo.NewMongoDBUserRepository(db)
+
+	// Auth Use case
+	smsProvider := twilio.NewTwilioSMSProvider(twilioClient)
+	authUseCase = userUC.NewAuthUseCase(userRepo, smsProvider)
+}
+
+func addRoutes(e *echo.Echo) {
```
Comment (YeongZhenHong) [inline smashsync-go/cmd/rest/main.go:96]: Missing comments on routes

File: smashsync-go/cmd/rest/main.go [hunk 3]
```diff
@@ -1,15 +1,108 @@
...
+	e.GET("/health", func(c echo.Context) error {
```
Comment (YeongZhenHong) [inline smashsync-go/cmd/rest/main.go:97]: Avoid using get, if possible use post over get as get is less secure

**Review by YeongZhenHong — CHANGES_REQUESTED** [review]

File: smashsync-go/cmd/rest/main.go (hunk 2 above)
Comment (dubstep-warrior) [reply smashsync-go/cmd/rest/main.go:96]: Updated with comments

File: smashsync-go/cmd/rest/main.go (hunk 3 above)
Comment (dubstep-warrior) [reply smashsync-go/cmd/rest/main.go:98]: Updated Auth endpoints, handlers and corresponding swagger docs.

File: smashsync-go/cmd/rest/main.go (hunk 1 above)
Comment (dubstep-warrior) [reply smashsync-go/cmd/rest/main.go:33]: Updated with comments

File: smashsync-go/docs/swagger.json [hunk 4]
```diff
@@ -1,7 +1,90 @@
...
+                        "schema": {
+                            "type": "string"
+                        }
+                    }
+                }
+            },
+            "post": {
+                "description": "Authenticate a user based on the provided OTP code.",
+                "consumes": [
+                    "multipart/form-data"
```
Comment (YeongZhenHong) [inline smashsync-go/docs/swagger.json:48]: JSON as request payload or form data? Inconsistent request types

File: smashsync-go/cmd/rest/main.go (hunk 3 above)
Comment (YeongZhenHong) [reply smashsync-go/cmd/rest/main.go:103]: JSON as request payload or form data?

**Review by YeongZhenHong — CHANGES_REQUESTED** [review]

File: smashsync-go/docs/swagger.json (hunk 4 above)
Comment (dubstep-warrior) [reply smashsync-go/docs/swagger.json]: Updated to JSON payload

**Review by YeongZhenHong — APPROVED** [review]

**Pull request merged**